### PR TITLE
spg Chapter Five

### PR DESCRIPTION
**Pull request by SimonGross**
Chapter Five answers

### CODE REVIEW

File: src/com/datacash/training/Bubble.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		System.out.println( "Sorted array is:" );
+		for ( int n = 0 ; n < nums.length ; n++ ) {
+			System.out.print( " " + nums[n] );
+		}
+		System.out.println();
+
+	}
+	
+	private static void bubbleSortList() {
+		List<Integer> nums = Arrays.asList( 99, -10, 100123, 18, -978, 5623, 463, -9, 287, 49 );
```
Comment (alexdeas) [inline src/com/datacash/training/Bubble.java:52]: So is using Lists faster than Arrays or what?

Comment (utopia-2) [inline src/com/datacash/training/Bubble.java:52]: Due to method overhead, slightly slower, @alexdeas, but `List` has the `stream` method :grinning:.

File: src/com/datacash/training/Question7.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+		System.out.print( "Original message: " );
+		System.out.println( msg );
+
+		msg.chars().forEach( ( ch ) -> encMsg.append( (char) (ch ^ key.charAt( msg.indexOf( ch ) % 8 )) ) );
+
+		System.out.print( "Encoded message: " );
+		System.out.println( encMsg );
+
+		encMsg.chars().forEach( ( ch ) -> decMsg.append( (char) (ch ^ key.charAt( encMsg.indexOf( Character.toString( (char) ch ) ) % 8 )) ) );
```
Comment (alexdeas) [inline src/com/datacash/training/Question7.java:21]: What's with the turning the key character back into a string? It should be possible to decode with the same operations as encoding.

File: src/com/datacash/training/chapter5.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+11) Both these cases will cause an error, as the boundaries of an array are strictly enforced
+
+12) >>>
+
+13) See Question13.java
+
+14) The for loops cannot be converted into for-each loops, as the loop control variables are required to assign values to the area when sorting. It is possible to do this when using Lists as demonstrated in Question 4, although it is not as efficient.
+
+15) As of Java 7, switch statements can be controlled by a String.
```
Comment (alexdeas) [inline src/com/datacash/training/chapter5.md:37]: Correct answers!

File: src/com/datacash/training/Question7.java (hunk 2 above)
Comment (SimonGross) [inline src/com/datacash/training/Question7.java:21]: For whatever reason `StringBuilder.indexOf` will only accept a String, not a char. In hindsight I could have called `toString` at some point to make sure the same operations were being performed.

File: src/com/datacash/training/chapter5.md (hunk 3 above)
Comment (codavex) [inline src/com/datacash/training/chapter5.md:37]: Agreed!

Comment (SimonGross) [inline src/com/datacash/training/chapter5.md:37]: :tada:

**Pull request closed without merging**